### ISSUE TITLE
Feature request : record user's sessions

### ISSUE THREAD

**Issue by mldmld68** [GoogleCloudPlatform/iap-desktop#546]
Hi,

For security reasons, one need to record the users's sessions.

It would be great if we could setup record sessions (.mp4 for Windows, text file for linux) on a GCS with the userID used to open the session as metadata or in the file name.

Combined with our Citrix platform, one can build a very effective bastion

Thanks for you consideration
Regards
MLD

**Comment by jpassing** [comment]
I see the value of session recording, but I think adding recording capabilities to IAP Desktop might not be too useful: If the client (i.e., IAP Desktop) does the recording, then it's difficult to prevent users from tampering with, or disabling the recording. To make sure that recordings are always enabled and complete, they'd have to be created server side -- either by IAP TCP Forwarding or by a bastion host. 
Would you agree with that, or is there a reason why you think the recording should be done client-side?

**Comment by mldmld68** [comment]
Yes, a user could disable the recording if iap-desktop runs on their local laptop
But we could offer the service through (and only through) a Citrix infrastructure to provide the access to iap-desktop.
iap-desktop would run on the Citrix servers and will be streamed to the user's laptop.

The Citrix server would have the TCP ports opened to access IAP and iap-desktop use an identity allowed to store records's files on a GCS

**Comment by smartyr73** [comment]
In terms of Citrix session recording, this is something I already do with Citrix apps (So it is possible) and I've exactly the same use case in mind.

**Comment by ernestosantander-Rnlt** [comment]
hi, is there any update on this request ? I see other (very well known ..) CSPs already providing this feature !! 
thank you

**Comment by jpassing** [comment]
As far as IAP TCP forwarding is concerned, the situation has not changed: IAP-TCP is protocol-agnostic -- on the positive side, that means you can use it for any TCP-based protocol, not just SSH and RDP. But on the negative side, it also means that IAP-TCP doesn't "understand" SSH and RDP, and doesn't provide capabilities like traffic inspection or session recording. While this may change in the future, I'm currenly not aware of any such plans.
